ViewControllers/MostPopularCollectionViewController.swift

```diff
@@ -91,16 +82,34 @@ class MostPopularViewController: UIViewController, UICollectionViewDelegate {
         button.translatesAutoresizingMaskIntoConstraints = false
         button.setTitle("See more", for: .normal)
         button.titleLabel?.font = UIFont.systemFont(ofSize: 14, weight: .semibold)
-        button.tintColor = UIColor(named: "acidGreen")
+        button.tintColor = UIColor(named: "titleColor")
         button.alpha = 0
         return button
     }()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        title = "Most popular"
-        navigationController?.navigationBar.titleTextAttributes = [.foregroundColor: UIColor.white, .font: UIFont.systemFont(ofSize: 24, weight: .heavy)]
+        setupUI()
+        
+        movieDataService.getMostPopularMoviesList { [weak self] (moviesList: [Movie]?) in
+            if let moviesList = moviesList {
+                self?.movies = moviesList
+            }
+            
+            DispatchQueue.main.async { [weak self] in
+                self?.collectionView.reloadData()
+                self?.itemInViewIndex = Int(round(Double((self?.movies.count)! / 2)))
+                self?.collectionView.scrollToItem(at: IndexPath(item: self!.itemInViewIndex, section: 0), at: UICollectionView.ScrollPosition.centeredHorizontally, animated: false)
+                self?.configureWithData(index: self!.itemInViewIndex)
+                self?.fadeIn()
+                }
```

Comment:
Since we are safely checking for `self` existence in the lines above we don't need to keep the optional self here. Also, this block is not going to retain self anymore so we don't need this second `[weak self]`
```suggestion
            DispatchQueue.main.async {
                self.collectionView.reloadData()
                self.itemInViewIndex = Int(round(Double(self.movies.count / 2)))
                self.collectionView.scrollToItem(at: IndexPath(item: self.itemInViewIndex, section: 0), at: UICollectionView.ScrollPosition.centeredHorizontally, animated: false)
                self.configureWithData(index: self.itemInViewIndex)
                self.fadeIn()
                }
```